flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/config/FlinkConfigBuilder.java

```diff
@@ -55,31 +54,45 @@
 
 /** Builder to get effective flink config from {@link FlinkDeployment}. */
 public class FlinkConfigBuilder {
-    private final ObjectMeta meta;
+    private final String namespace;
+    private final String clusterId;
     private final FlinkDeploymentSpec spec;
     private final Configuration effectiveConfig;
-
-    public static final Duration DEFAULT_CHECKPOINTING_INTERVAL = Duration.ofMinutes(5);
-
-    public FlinkConfigBuilder(FlinkDeployment deploy, Configuration flinkConfig) {
-        this(deploy.getMetadata(), deploy.getSpec(), flinkConfig);
+    private final boolean forDeployment;
+
+    protected static final Duration DEFAULT_CHECKPOINTING_INTERVAL = Duration.ofMinutes(5);
+
+    protected FlinkConfigBuilder(
+            FlinkDeployment deployment, Configuration flinkConfig, boolean forDeployment) {
+        this(
+                deployment.getMetadata().getNamespace(),
+                deployment.getMetadata().getName(),
+                deployment.getSpec(),
+                flinkConfig,
+                forDeployment);
     }
 
-    public FlinkConfigBuilder(
-            ObjectMeta metadata, FlinkDeploymentSpec spec, Configuration flinkConfig) {
-        this.meta = metadata;
+    protected FlinkConfigBuilder(
+            String namespace,
+            String clusterId,
+            FlinkDeploymentSpec spec,
+            Configuration flinkConfig,
+            boolean forDeployment) {
+        this.namespace = namespace;
+        this.clusterId = clusterId;
         this.spec = spec;
         this.effectiveConfig = new Configuration(flinkConfig);
+        this.forDeployment = forDeployment;
```

Comment:
Do we have to add comments for `forDeployment`. It seems mainly to save the create temp template file Right ?

Reply:
I removed this already